src/private/impls.rs

```diff
@@ -1,25 +1,50 @@
-use super::{
-    Bits,
-    FromBits,
-    IntoBits,
-    PopBits,
-    PushBits,
+use crate::{
+    error::{
+        InvalidBitPattern,
+        OutOfBounds,
+    },
+    Specifier,
 };
-use crate::Specifier;
 
 impl Specifier for bool {
     const BITS: usize = 1;
-    type Base = u8;
-    type Face = bool;
+    type Bytes = u8;
+    type InOut = bool;
+
+    #[inline]
+    fn into_bytes(input: Self::InOut) -> Result<Self::Bytes, OutOfBounds> {
+        Ok(input as u8)
+    }
+
+    #[inline]
+    fn from_bytes(
+        bytes: Self::Bytes,
+    ) -> Result<Self::InOut, InvalidBitPattern<Self::Bytes>> {
+        match bytes {
+            0 => Ok(false),
+            1 => Ok(true),
+            invalid_bytes => Err(InvalidBitPattern { invalid_bytes }),
```

Comment:
Could this be an immediate panic, since `from_bytes` would only ever be called with a single masked bit?

Reply:
I understand your point and you are right that if this `Err` is returned it will very likely (or always) be a bug in the implementation. I chose to return the `Err` still because I wanted to mirror the expectation of the API and also performance wise it shouldn't make a difference due to heavy Rust compiler optimizations.

Reply:
Seems reasonable, yeah it should optimize out either way.